### ISSUE TITLE
Self dispatched workflow functions

### ISSUE THREAD

**Issue by Keyrxng** [ubiquity-os-marketplace/ubiquity-os-kernel-telegram#20]
Now that the kernel' `APP_ID` and `APP_PRIVATE_KEY` have been exposed as org secrets I think I should go back to my original implementation where the worker for this plugin controlled the dispatching of it's own workflows for MTProto API (workrooms, etc...) features.

- with two config entries ubiquity-os kernel is dispatching tons of workflows for no reason which can be avoided
- self-dispatched means we can validate no-op payloads first
- it's cleaner and more efficient from a plugin standpoint
- removes the need for two entries in the config
- optimizes the github rate limit consumed by this plugin
- it's no longer considered "unsafe" to do so or the secrets wouldn't be exposed

**Comment by 0x4007** [comment]
This will need a time estimate as well.

**Comment by Keyrxng** [comment]
I cant assign labels

Probs about a day to rewrite and test everything is working again

**Comment by Keyrxng** [comment]
@0x4007 I'm not free today to carry out work but will be over the next day or two as a heads up

Thanks for labeling for me.

**Comment by whilefoo** [comment]
A plugin should ideally not dispatch workflows on the behalf of the kernel as that is the job of the kernel. Can't we move everything to the workflow?

**Comment by Keyrxng** [comment]
Not as far as I'm aware @whilefoo no.

We need to set a delivery url for our BotFather bot updates and so we use our CF worker domain. What would we set it to if using actions? Afaik that's not possible but I'm no expert.  

---

I see recent discussion regarding "hybrids" and plugins being able to self-dispatch in [another task](https://github.com/ubiquity-os-marketplace/daemon-pricing/pull/72#issuecomment-2591918348).

I don't feel the kernel needs to gatekeep the ability to dispatch workflows as it has other important jobs too but actually it's an improvement if plugins can self-dispatch for many reasons:

1. Worker as P.O.E means we can detect no-op payloads and optimize our rate limit usage which seems to have been a problem as of recent.
2. We have both instant and long-running capabilities for plugins.
3. With [Tail Workers](https://developers.cloudflare.com/workers/observability/logs/tail-workers/) we could setup a decent redundancy system for all worker/hybrid plugins.

So if the kernel handles handshakes properly and the plugins are reviewed that self-dispatch and/or leverage `APP_ID` and `APP_PRIVATE_KEY` then it's safe, no?

**Comment by Keyrxng** [comment]
Should I proceed with this @0x4007? I'm assuming yeah, but just clarifying before I get stuck in.

**Comment by Keyrxng** [comment]
Just waiting for my other active PRs in end-stage review to progress (hopefully today) and I'll submit a new PR for review here. 

I don't want to overload anyone or be seen as stretched-too-thin lmao, cheers.

**Comment by Keyrxng** [comment]
@gentlementlegen ty, the worker deployed https://ubiquity-os-telegram-bridge-development.ubiquity.workers.dev/

As I don't have the access I can't confirm whether it's using the `TEMP_SAFE_PAT` that I set in cloudflare directly or if it's using the private key correctly that should have been uploaded direct from org secrets.

Idk if we wanna sort me with temp access or if we can tag team it or something and see what's going on?

**Comment by gentlementlegen** [comment]
It is interesting that I didn't get any rewards for the review, I think this can be considered as a bug.

I'll have a look and come back to you.

**Comment by Keyrxng** [comment]
@gentlementlegen it needs re-run anyway as I never got the task reward dude and ty.

I should only need to update my env vars locally, paste them into the repo settings and re-run `worker-deploy` to update things which should take me a minute or less whenever it needs done to transfer onto the new bot and account. I hoped to confirm everything via the current accounts first before I put it on the new ones.

I will submit the accounts that I made for development as the staging branch details.

What needs updated:

- webhook url
- db url (not until after #53)
- admins
- api hash and id
- bot token
